lymph/unilateral.py

```diff
@@ -277,15 +281,15 @@ def tumor_spread_probs(self):
         values, the transition matrix - if it was precomputed - is deleted
         so it can be recomputed with the new parameters.
         """
-        return np.array([edge.spread_prob for edge in self.tumor_edges], dtype=float)
+        return np.array([edge.get_spread_prob for edge in self.tumor_edges], dtype=float)
 
     @tumor_spread_probs.setter
     def tumor_spread_probs(self, new_tumor_probs):
         """Set the spread probabilities for the connections from the tumor to
         the LNLs.
         """
-        for i, edge in enumerate(self.tumor_edges):
-            edge.spread_prob = new_tumor_probs[i]
+        for edge, spread_prob in zip(self.tumor_edges, new_tumor_probs):
+            edge.set_spread_prob = spread_prob

```

Comment:
Also here: Since you do `spread_prob = property(...)` in the `Edge` class, you can continue to use it as if you had used the `@property` decorator.

Of course its also fine to use `edge.set_spread_prob(new_spread_prob)`, but what you do here is wrong, I think: You are replacing the `set_spread_prob` method with a number. Also in the getter, which now returns an array of methods, not the actual numbers.